security/pkg/nodeagent/vm/citadelagent_v2.go

```diff
@@ -59,15 +61,22 @@ const (
 	projectIDField     = "project_id"
 )
 
+// Error messages.
+const (
+	nilConfigErr      = "node Agent configuration is nil"
+	wrongPlatformErr  = "node Agent is not running on the right platform"
+	maximumRetiresErr = "node agent can't get the CSR approved from Istio CA after max number of retries (%d)"
```

Comment:
I think you can just do inline fmt.Errorf("", %v"), fmt.Error("").

In the test, you just do strings.Contain(err, "critical sub string")

Reply:
Is there a convention? I think it's less error prone this way.

Reply:
each of your error string fmt may or may not have `%v` positional argument.

When they're used, it's just as fmt.Error or fmt.Errorf(stringVar, argument). Not clear by looking whether it's a misusage or or not.

Another good practice is in general, test data should be more inline, not more indirection when possible. Such that the test itself is easier to understand.

Reply:
I may have misinterpreted your comment. To sum up, which one do you prefer?
```
someErrMsg = "some error message"
...
fmt.Errorf(fmt.Sprintf("%s %d", someErrMsg, someNumber))
// And use someErrMsg inside the test as well.
```
or 
```
fmt.Errorf("some error message %d", someNumber)
// Inside the test, use some unique substring of "some error message"
```

Reply:
Done.